### DISCUSSION TITLE
In Porous flow, can we use kernels instead of action to 'save_component_rate_in'?

### DISCUSSION THREAD

**Discussion by robo-boluo** [idaholab/moose#27550] [Q&A Modules: Porous Flow] [answered]
Dear Moose community,
I followed the coupled flow example under the folder of '(modules/combined/examples/geochem-porous_flow/geotes_weber_tensleep/porous_flow.i). The example uses '[PorousFlowFullySaturated]' action to save the mass fraction rate. 

![WX20240505-132012](https://github.com/idaholab/moose/assets/59897601/bd8b6cb5-76be-460c-b5b0-3d52e2e22bd0)

Considering the extension of multiple phases and unsaturated flow, my question is that can we use kernels such as, [PorousFlowMassTimeDerivative] to save the mass fraction rate? 

Thanks for your time and help.

Robo

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can look at PorousFlowFullySaturated.C to see what this parameter is doing. 
```
      kernel_name = "PorousFlowFullySaturated_MassTimeDerivative" + Moose::stringify(i);
      params.set<unsigned int>("fluid_component") = i;
      params.set<NonlinearVariableName>("variable") = _mass_fraction_vars[i];
      if (_save_component_rate_in.size() != 0)
        params.set<std::vector<AuxVariableName>>("save_in") = {_save_component_rate_in[i]};
      _problem->addKernel(kernel_type, kernel_name, params);
```

it is just using the "save_in" parameter on the kernels. You can do that too with an input file that uses the kernels instead of the action

Guillaume

**Comment by robo-boluo** [comment]
many thanks for the prompt reply.
